internal/compiler/passes/lower_component_container.rs

```diff
@@ -29,20 +34,46 @@ fn diagnose_component_container(element: &ElementRc, diag: &mut BuildDiagnostics
     }
 }
 
-fn process_component_container(element: &ElementRc, empty_type: &ElementType) {
+fn process_component_container(
+    element: &ElementRc,
+    empty_type: &ElementType,
+    new_components: &mut Vec<Rc<Component>>,
+) {
+    let component = Rc::new_cyclic(|component_weak| {
+        let root_element = Rc::new(RefCell::new(Element {
+            id: smol_str::format_smolstr!("__cc_root__{}", new_components.len()),
+            base_type: empty_type.clone(),
+            enclosing_component: component_weak.clone(),
+            ..Default::default()
+        }));
+
+        Component {
+            node: element.borrow().debug.first().map(|n| n.node.clone().into()),
+            id: smol_str::format_smolstr!("__CCC_{}", new_components.len()),
+            root_element,
+            ..Default::default()
+        }
+    });
+
     let mut elem = element.borrow_mut();
 
     let embedded_element = Element::make_rc(Element {
-        base_type: empty_type.clone(),
-        id: elem.id.clone(),
+        base_type: ElementType::Component(component.clone()),
         debug: elem.debug.clone(),
         enclosing_component: elem.enclosing_component.clone(),
         default_fill_parent: (true, true),
-        is_legacy_syntax: elem.is_legacy_syntax,
         inline_depth: elem.inline_depth,
+        repeated: Some(RepeatedElementInfo {
+            model: expression_tree::Expression::BoolLiteral(false),
```

Comment:
Should that not be Expression::Invalid or something? The model should never be executed.
I can imagine a future pass that optimizes repeater element which are always false.

Reply:
When I started out, I suspected this might end up getting used, so I wanted to be careful.

Any element with `is_component_placeholder` is making `can_optimize` return `false` in the `optimize_useless_rectangles` pass... but of course future passes might decide to have different logic.

I'll see whether this still works when I use `Expression::Invalid`.

Reply:
This panics the compiler when lowering expressions. In general that makes a lot of sense IMHO, so I do not want to meddle with this right now.